packages/block-editor/src/store/index.js

```diff
@@ -17,11 +17,15 @@ import controls from './controls';
  */
 const MODULE_KEY = 'core/block-editor';
 
-const store = registerStore( MODULE_KEY, {
+export const storeConfig = {
```

Comment:
An alternative here to exposing all the implementation details (which, on the surface, is not exceedingly worrisome) is, as in #7453, exposing a `createStore` which registers the store given a registry (and I suppose some optional options object for the default case in applying the `persist` option).

https://github.com/WordPress/gutenberg/pull/7453/files#diff-adc2d694b934c0d37f4ae99f4595afc8L26

Reply:
My ideal API is this:

```js
const store1Config = {};
const store2Config = {};
const store1 = createStore( store1Config );
const store2 = createStore( store2Config );
const registry = createRegistry( [ store1, store2 ] );
const store3 = createStore( store2Config );
const subRegistry = createRegistry([ store1, store3 ]);
```

Unfortunately, this is not possible because the current `createStore` is dependent on the `registry` it's being used in. I explored whether we can completely separate the two but it's hard to do with the namespace stores.

--- 
This doesn't reply directly to your comment I guess but yeah, I feel that sharing configs is the way to go in general and registering should be part of the components mounting... The global registry makes this hard to achieve at the moment.